src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -22,57 +22,37 @@
 namespace stellar
 {
 
-ApplyLedgerChainWork::ApplyLedgerChainWork(
-    Application& app, TmpDir const& downloadDir, LedgerRange range,
-    LedgerHeaderHistoryEntry& lastApplied)
-    : BasicWork(app, "apply-ledger-chain", RETRY_NEVER)
+ApplyCheckpointWork::ApplyCheckpointWork(Application& app,
+                                         TmpDir const& downloadDir,
+                                         uint32_t checkpoint, LedgerRange range)
```

Comment:
This look strange to me. Why `ApplyCheckpointWork` need to know the whole range of applied checkpoint? I'm assuming that this is the responsibility of `DownloadApplyTxsWork`.

It looks like there is new condition with mRange added: `auto done = lcl == mRange.last() || lcl == mCheckpoint;`. This is probably for applying the last checkpoint, which may not be applied in full in case of command line catchup with `catchup-to` not pointing to a checkpoint end, as each `ApplyCheckpointWork` gets exactly the same `mRange` from `DownloadApplyTxsWork`.

It seem like a little bit better API could be used here. I see two possibilities:
1. replace `LegerRange range` with `uint32_t boundary` that is equal to `range.last`
2. replace pair of `uint32_t checkpoint, LedgerRange range` with a new `LedgerRange range` that has both ends (`first` and `last`) in the same checkpoint (which is obviously the just removed `uint32_t checkpoint`).